conans/client/graph/compatibility.py

```diff
@@ -106,13 +107,15 @@ def _should_migrate_file(file_path):
 
 class BinaryCompatibility:
 
-    def __init__(self, compatibility_plugin_folder):
+    def __init__(self, cache_folder):
+        compatibility_plugin_folder = HomePaths(cache_folder).compatibility_plugin_path
         compatibility_file = os.path.join(compatibility_plugin_folder, "compatibility.py")
         if not os.path.exists(compatibility_file):
             raise ConanException("The 'compatibility.py' plugin file doesn't exist. If you want "
                                  "to disable it, edit its contents instead of removing it")
         mod, _ = load_python_file(compatibility_file)
         self._compatibility = mod.compatibility
+        self._hook_manager = HookManager(HomePaths(cache_folder).hooks_path)
```

Comment:
You shouldn't instantiate ``HookManager`` here anymore.

Reply:
Done, consuming hook_manager from conan_api.config.hook_manager instead. Please, see the commit 59c6a83d3